src/bin/pgcopydb/cli_common.c

```diff
@@ -171,9 +171,9 @@ cli_copydb_getenv(CopyDBOptions *options)
 	/* now some of the options can be also set from the environment */
 	if (env_exists(PGCOPYDB_SOURCE_PGURI))
 	{
-		if (!get_env_copy(PGCOPYDB_SOURCE_PGURI,
-						  options->source_pguri,
-						  sizeof(options->source_pguri)))
+		if (!get_env_copy_dynamic(PGCOPYDB_SOURCE_PGURI,
+						  &(options->source_pguri),
+						  NULL))
```

Comment:
That feels strange. I suppose I would have gone with `get_env_dup` that calls `strdup` instead of `strcpy`, and I would arrange to use the same preliminary code from `get_env_copy_with_fallback`. Also we might have to introduce the variant with and without a fallback value.